修正在售房屋列表更新后初次发现时间不正确的问题；修正已经售出房屋不会删除的问题

jim-kirisame opened the pull request.

jim-kirisame committed:
修正在售房屋列表更新后初次发现时间不正确的问题；修正已经售出房屋不会删除的问题

jim-kirisame committed:
使用服务器时间代替本地时间

jim-kirisame committed:
修正时间是UTC的问题

jim-kirisame committed:
Bump version

jim-kirisame committed:
fix async hang [Bluefissure]

Signed-off-by: Jim Jiang <jim@lotlab.org>

jim-kirisame committed:
更新opcode

jim-kirisame committed:
修正自动上报快照失败后，无法手动上报快照的问题

jim-kirisame committed:
添加TTS功能；服务器ID与解析插件脱钩

jim-kirisame committed:
添加弹出通知功能，使用Fody.Costura打包DLL

jim-kirisame committed:
Opcode to 5.55

hyperzlib commented:
虽然来晚了

那个已售出的不移除是特性，下次打开act的时候会移除，另外我个人觉得TTS应该用触发器实现，避免卡住ACT。其他功能会尽量合并。

jim-kirisame commented:
哦草怎么后续的提交也开PR了

顺带问一句 TTS会卡住ACT的吗？

hyperzlib commented:
看怎么处理的，反正输出了日志行，一些操作可以交给触发器做

另外我后来发现查房会卡和是不是使用异步没有关系，所以后来还是用同步处理了，效率反而比异步高，还不会死锁

jim-kirisame commented:
据我这边的观察，同步确实会在空房的时候卡，不过具体我也没仔细看是什么原因卡住的

hyperzlib commented:
空房会卡主要是写入日志行的原因，无论是异步还是同步，写入日志行都是经过ACT的同步消息队列处理的，所以区别不是很大。

另外，那个搬家查询网站是你做的吗？算是实现了我想做但是没空做的功能。因为神拳痕房子多，都没什么动力开发查房了。

jim-kirisame commented:
nga的那个吗？不是我做的。我做的是另一个，依赖于这个插件做上报的，所以才会有上面这些改动。

hyperzlib commented:
另外，基于ACT的Log输出，可以做个悬浮窗，方便看一些。

hyperzlib commented:
嗯……有办法只合并一部分吗？说实话我还没怎么用过github的merge功能

jim-kirisame commented:
残念，没有。你要合并哪个部分我手动给你cheerypick重新开pr吧

hyperzlib commented:
对了，那个自动上报的问题，是指如果只有空房变成已售出不会上报吗？

jim-kirisame commented:
不是，主要是为了解决部分情况下手动上报无法上报快照的问题

hyperzlib commented:
明白了，是指即使快照内容没有更新，通过手动更新也会上传

hyperzlib commented:
这个插件以后我应该不会做大的更新了（除非转回紫水），之前想加的一个功能是选中房屋右侧显示小地图，方便看位置（这个是二区专用功能）

hyperzlib commented:
另外如果有服务器的话，最好能在线更新opcode，因为github连通性不好，所以没做，不过用gitee应该可以。

jim-kirisame commented:
看小地图功能我这边的网站上倒是有；在线更新opcode这个我个人不是特别喜欢在线热更新所以做的动力也不高

hyperzlib commented:
也不是热更新，就是opcode从配置文件读取，如果通过http获取了新的opcode就写入配置文件

因为不在国服的插件市场里，更新挺麻烦的

还有，房区ID也可以先加到热更新里，为6.0的雪景房做准备

jim-kirisame committed:
调整数据解析流程，降低卡顿

- 不再将解析流程放入UI线程中以免造成解析卡住UI的问题
- 使用AppendText代替原有的字符串运算，解决了日志行过多时写入日志过慢的问题
- 使用BeginInvoke代替Invoke异步更新UI以免UI更新阻塞解析数据
- TTS与通知弹出使用异步Task，以免阻塞解析数据

jim-kirisame committed:
更新版本号

jim-kirisame committed:
为房屋快照加锁，以规避多线程下可能出现的问题
